supplementary/alica_ros1/alica_ros_turtlesim/libalica-turtlesim/src/SpawnTurtle.cpp

```diff
@@ -0,0 +1,39 @@
+#include "SpawnTurtle.h"
+
+#include <ros/ros.h>
+#include <turtlesim/Spawn.h>
+
+#include <memory>
+#include <random>
+
+namespace turtlesim
+{
+
+SpawnTurtle::SpawnTurtle(alica::BehaviourContext& context)
+        : alica::BasicBehaviour(context)
+{
+}
+
+void SpawnTurtle::run()
+{
+    ros::ServiceClient spawnSrvClient = ros::NodeHandle("~").serviceClient<turtlesim::Spawn>("/spawn");
+    turtlesim::Spawn spawnSrv;
+    spawnSrv.request.x = 1;
+    spawnSrv.request.y = 1;
+    spawnSrv.request.theta = 0;
+    spawnSrv.request.name = alica::LockedBlackboardRO(*getGlobalBlackboard()).get<std::string>("agentName");
+    if (spawnSrvClient.waitForExistence() && spawnSrvClient.call(spawnSrv)) {
```

Comment:
probably should not wait forever?

Reply:
found an exists() method that does not block I think. Will use that instead

Reply:
looks like call() blocks anyway, so I'm going to leave this as is